### PR TITLE
Using SNS to trigger canary via lambda

### PR DESCRIPTION
**Pull request by laurence-bird**
Removing TriggeredEventPublisher, and now triggering events via the lambda from comms-example-trigger

### CODE REVIEW

File: conf/prd.conf [hunk 1]
```diff
@@ -11,6 +11,13 @@ kafka {
   bootstrap.servers = "@@{prd.kafka_hosts}"
 }
 
+aws {
+  region = "eu-west-1"
+  canary {
+    sns_topic = ${example_trigger.sns_topic}
```
Comment (fleetwoods) [inline conf/prd.conf]: @@{prd.example_trigger.sns_topic}

File: conf/uat.conf [hunk 2]
```diff
@@ -7,6 +7,13 @@ librato {
   }
 }
 
+aws {
+  region = "eu-west-1"
+  canary {
+    sns_topic = ${example_trigger.sns_topic}
```
Comment (fleetwoods) [inline conf/uat.conf]: @@{uat.example_trigger.sns_topic}

File: src/main/scala/com/ovoenergy/comms/monitor/canary/email/client/SNS.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.amazonaws.services.sns.{AmazonSNS, AmazonSNSClient}
+import com.typesafe.config.ConfigFactory
+
+import scala.util.Try
+
+object SNS {
+  private val config    = ConfigFactory.load()
+  private val region    = config.getString("aws.region")
+  private val snsTopic  = config.getString("aws.canary.sns_topic")
+
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/comms/monitor/canary/email/client/SNS.scala]: We have been managing all config in Main, and currying, do we want to keep this consistent in the project?

File: src/main/scala/com/ovoenergy/comms/monitor/canary/email/trigger/CanaryEmailTrigger.scala [hunk 4]
```diff
@@ -1,40 +1,15 @@
...
-        customerId = "canary",
-        traceToken = traceToken,
-        commManifest = CommManifest(commType = Service, name = "canary", version = "0.1"),
-        friendlyDescription = "Canary event for email monitoring",
-        source = "comms-monitoring-service",
-        canary = true,
-        sourceMetadata = None
-      )
-    }
+  def apply(traceToken: String): Unit = {
```
Comment (fleetwoods) [inline src/main/scala/com/ovoenergy/comms/monitor/canary/email/trigger/CanaryEmailTrigger.scala]: Probably just lose this?

Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/monitor/canary/email/trigger/CanaryEmailTrigger.scala]: oops yeah forgot to remove, thanks

File: src/main/resources/application.conf [hunk 5]
```diff
@@ -7,6 +7,13 @@ librato {
   }
 }
 
+aws {
+  region = "eu-west-1"
+  canary {
+    sns_topic = ${example_trigger.sns_topic}
```
Comment (cb372) [inline src/main/resources/application.conf]: I would make this consistent with the usual naming for environment variables: upper case, no dots. e.g. `${EXAMPLE_TRIGGER_SNS_TOPIC}`

File: src/main/scala/com/ovoenergy/comms/monitor/Main.scala [hunk 6]
```diff
@@ -51,15 +48,18 @@ object Main extends App {
...
+  val kafkaConsumerGroupId    = config.getString("kafka.group.id")
+  val kafkaBootstrapServers   = config.getString("kafka.bootstrap.servers")
 
   val environment             = sys.env.getOrElse("ENV", throw new IllegalStateException("ENV environment variable required"))
   val communicationPipelineTimeout = FiniteDuration(config.getLong("pipeline.timeout.seconds"), "seconds")
 
   val libratoEmail            = config.getString("librato.api.email")
   val libratoToken            = config.getString("librato.api.token")
 
+  val awsRegion                  = config.getString("aws.region")
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/monitor/Main.scala]: If you're going to be all OCD about aligning your equals signs, you could at least put the right number of spaces 😛

File: src/main/scala/com/ovoenergy/comms/monitor/canary/email/client/SNS.scala [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.sns.model.PublishResult
+import com.amazonaws.services.sns.{AmazonSNS, AmazonSNSClient}
+import com.typesafe.config.ConfigFactory
+
+import scala.util.Try
+
+object SNS {
+  private val config    = ConfigFactory.load()
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/monitor/canary/email/client/SNS.scala]: not used

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 121](https://circleci.com/gh/ovotech/comms-monitor-service/121) at 2017-01-03 15:27 GMT.

**Pull request merged**